activation/post.go

```diff
@@ -11,9 +12,12 @@ import (
 	"github.com/spacemeshos/post/proving"
 
 	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/datastore"
 	"github.com/spacemeshos/go-spacemesh/log"
 )
 
+//go:generate mockgen -package=activation -destination=./post_mock.go -source=./post.go
```

Comment:
mocks are generated in the mocks directory. please do it consistently with the rest

Reply:
That's not true, there are mocks in the `activations` directory, at least one of which I moved with this PR to the `mocks` directory.

I can move mocks for this file to the mock directory too, but then I run into cyclic dependencies that are resolved by moving all types in this file into the `common/types` package. That is imho where they belong anyway, but will cause the PR to become even bigger ;-)

Reply:
refactored package to have mock in its own package.